job-cleanup/cleanup.py

```diff
@@ -15,27 +15,31 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
-from __future__ import print_function
-
+import logging
 import os
 import socket
 import sys
 import time
 
-from kubernetes import KubernetesAPIClient
+from typing import Tuple
 
+from dotmap import DotMap
+from tiny_kubernetes import KubernetesAPIClient
 
 NAMESPACE = '/var/run/secrets/kubernetes.io/serviceaccount/namespace'
 TIMEOUT = float(os.environ.get('WAIT_TIMEOUT', '10'))
-RETRIES = int(os.environ.get('WAIT_RETRIES', '24'))
+RETRIES = int(os.environ.get('WAIT_RETRIES', '5'))
 RETRY_DELAY = float(os.environ.get('WAIT_DELAY', '5.0'))
 
-USE_KUBE_CONFIG = os.environ.get('USE_KUBE_CONFIG', False)
+LOG_LEVEL = logging.getLevelName(os.environ.get('LOG_LEVEL', 'INFO'))
+logging.basicConfig(level=LOG_LEVEL,
+                    format='%(asctime)-15s - %(levelname)-8s - %(message)s')
+logger = logging.getLogger(__name__)
 
 pod_is_self = True
 
 
-def get_current_namespace():
+def get_current_namespace() -> str:
```

Comment:
Why type cast everything now?

Reply:
They aren't type casts, they're [type hints](https://www.python.org/dev/peps/pep-0484/) for python 3.5. Decent IDEs (pycharm, etc) will warn about improper variable usage with them enabled.